telematic_system/telematic_cloud_messaging/src/main/java/com/telematic/telematic_cloud_messaging/message_converters/JSON2KeyValuePairsConverter.java

```diff
@@ -22,21 +22,29 @@ public String convertJson2KeyValuePairs(String json_str, List<String> to_str_val
             for (Object key : json.keySet()) {
                 key_count++;
                 Object value = json.get(key.toString());
+                boolean is_ignored = false;
 
                 if (value == null || value.toString().isEmpty())
                 {
                     pairs += key + "=\"NA\"";
                 }
                 else {
-                    boolean is_skip = false;
-                    for (String value_item: to_str_values)
+                    for(String field: ignore_fields)
                     {
-                        if (key.toString().strip().equals(value_item)){
+                        if(key.toString().strip().equalsIgnoreCase(field))   
+                        {
+                            is_ignored = true;
```

Comment:
Looks like this check should end once we find that the field should be ignored. Can we break after the condition is satisfied?

Reply:
The break here only break the inner loop not the JSON loop